src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -81,5 +100,62 @@ public AuthState CreateAuthState(HttpRequest httpRequest)
 
             return new AuthState(ExternalIdentifier, ExternalSystem);
         }
+
+        public async Task<AuthState> RetrieveAuthState(string nonce, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Get a reference to the blob
+                BlobClient blobClient = _blobContainerClient.GetBlobClient(nonce);
```

Comment:
I think we need to make sure that this is not null - OR - does this throw if `GetBlobClient()` is called with a nonce that does not exist?